src/language_features/hover.rs

```diff
@@ -105,43 +116,112 @@ pub fn editor_hover(
                     FACE_INFO_RULE, FACE_INFO_DEFAULT
                 )),
             HoverContents::Markup(contents) => match contents.kind {
-                MarkupKind::Markdown => markdown_to_kakoune_markup(contents.value, force_plaintext),
+                MarkupKind::Markdown => {
+                    if let HoverType::InfoInHoverClient { .. } = hover_type {
+                        contents.value
```

Comment:
yep, we need not interpret Markdown when we directly render it to a filetype=markdown file.

We should probably do the same in the other cases, where we get `MarkedString` objects

- `MarkedString::String` we can just pass on verbatim.
- `MarkedString::LanguageString` we should interpret as like this markdown snippet

    ```${language}
    ${value}
    ```